### PR TITLE
Sanitize docker image string

### PR DESCRIPTION
**Pull request by Tthomas63**
Fixes docker image text field being exploitable.
- Splits on spaces, and uses the first word
- Removes remaining white-spaces/edges
- Removes non-alphanumeric characters ( Besides those found in docker image names )

Utilizes regex, and string formatting.

### CODE REVIEW

File: codalab/codalabtools/compute/worker.py [hunk 1]
```diff
@@ -297,6 +298,12 @@ def run(task_id, task_args):
         detailed_results_url = task_args['detailed_results_url']
         private_output_url = task_args['private_output_url']
 
+        """
+        DOCKER IMAGE SANITATION
+        """
+        sant_docker_image = '"{}"'.format(docker_image.strip().split(' ')[0])
```
Comment (ckcollab) [inline codalab/codalabtools/compute/worker.py]: `sant_docker_image` -> `sanitized_docker_image`

Comment (ckcollab) [inline codalab/codalabtools/compute/worker.py]: And maybe describe what each step is trying to prevent?

Comment (Tthomas63) [inline codalab/codalabtools/compute/worker.py]: Okay. Wasn't sure about doing this since we are patching an exploit.

Comment (ckcollab) [inline codalab/codalabtools/compute/worker.py]: Good call, but I think we're OK here 🌮

Comment (Tthomas63) [inline codalab/codalabtools/compute/worker.py]: Sounds good!

**Comment by Tthomas63** [comment]
Requested changes made @ckcollab

File: codalab/codalabtools/compute/worker.py [hunk 2]
```diff
@@ -297,6 +298,16 @@ def run(task_id, task_args):
...
 
+        """
+        DOCKER IMAGE SANITATION
+        """
+
+        # Remove all excess whitespaces on edges, split on spaces and grab the first word.
+        sanitized_docker_image = '"{}"'.format(docker_image.strip().split(' ')[0])
+        # Regex acts as a whitelist here. Only alphanumerics and the following symbols are allowed: / . : -.
+        # If any not allowed are found, replaced with second argument to sub.
+        sanitized_docker_image = re.sub('[^0-9a-zA-Z/.:-]+', '', sanitized_docker_image)
```
Comment (ckcollab) [inline codalab/codalabtools/compute/worker.py]: As a fun practice exercise, let's add some tests for this -- we can pair program it tomorrow!

Comment (Tthomas63) [inline codalab/codalabtools/compute/worker.py]: Sounds good!

File: codalab/codalabtools/compute/worker.py [hunk 3]
```diff
@@ -297,6 +298,16 @@ def run(task_id, task_args):
         detailed_results_url = task_args['detailed_results_url']
         private_output_url = task_args['private_output_url']
 
+        """
+        DOCKER IMAGE SANITATION
+        """
+
+        # Remove all excess whitespaces on edges, split on spaces and grab the first word.
```
Comment (ckcollab) [inline codalab/codalabtools/compute/worker.py]: Also wraps in quotes which makes bash not interpret it, I believe?

**Comment by Tthomas63** [comment]
image

Now includes a jquery alert on maxLength as well to fix #1864 
*We might want to allow less characters, ~60*

**Pull request merged**